lib/eknvfs/ekn-file.c

```diff
@@ -0,0 +1,326 @@
+/* -*- mode: C; c-file-style: "gnu"; indent-tabs-mode: nil; -*- */
+/*
+ * ekn-file.c
+ *
+ * Copyright (C) 2016 Endless Mobile, Inc.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * Author: Juan Pablo Ugarte <ugarte@endlessm.com>
+ *
+ */
+
+#include "ekn-file.h"
+#include "ekn-file-input-stream-wrapper.h"
+
+#define EKN_SCHEME_LEN 6
+
+struct _EknFile
+{
+  GObject parent;
+};
+
+typedef struct
+{
+  gchar *uri;
+  EosShardBlob *blob;
+} EknFilePrivate;
+
+enum
+{
+  PROP_0,
+
+  PROP_URI,
+  PROP_BLOB,
+  N_PROPERTIES
+};
+
+static GParamSpec *properties[N_PROPERTIES];
+
+static void ekn_file_iface_init (gpointer g_iface, gpointer iface_data);
+
+G_DEFINE_TYPE_EXTENDED (EknFile,
+			ekn_file,
+			G_TYPE_OBJECT, 0,
+			G_ADD_PRIVATE (EknFile)
+			G_IMPLEMENT_INTERFACE (G_TYPE_FILE,
+                                               ekn_file_iface_init))
+
+#define EKN_FILE_PRIVATE(d) ((EknFilePrivate *) ekn_file_get_instance_private((EknFile*)d))
+
+static void
+ekn_file_init (EknFile *self)
+{
+
+}
+
+static void
+ekn_file_finalize (GObject *self)
+{
+  EknFilePrivate *priv = EKN_FILE_PRIVATE (self);
+
+  g_clear_pointer (&priv->uri, g_free);
+  g_clear_pointer (&priv->blob, eos_shard_blob_unref);
+  
+  G_OBJECT_CLASS (ekn_file_parent_class)->finalize (self);
+}
+
+static inline void
+ekn_file_set_uri (EknFile *self, const gchar *uri)
+{
+  EknFilePrivate *priv = EKN_FILE_PRIVATE (self);
+
+  g_return_if_fail (g_str_has_prefix (uri, "ekn:"));
+
+  if (g_strcmp0 (priv->uri, uri))
```

Comment:
Coding style: I do prefer being explicit on the condition, especially for `strcmp()`, instead of relying on the truthyness value of the integer result. In this case:

    if (g_strcmp0 (priv->uri, uri) != 0)